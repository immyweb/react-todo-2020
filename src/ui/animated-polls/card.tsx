import * as React from 'react';

import { APCard, APHeadline, APCopy, APCardImg, APCardCopy } from './styles';

import { ICard } from './types';

export const CardScreen: React.SFC<ICard> = ({ imageMobile, imageDesktop, title, copy }) => {
  return (
    <APCard>
      <APCardImg imageMobile={imageMobile} imageDesktop={imageDesktop} />
      <APCardCopy>
        <APHeadline>{title}</APHeadline>
        <APCopy>{copy}</APCopy>
      </APCardCopy>
    </APCard>
  );
};
