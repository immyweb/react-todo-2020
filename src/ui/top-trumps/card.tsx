import * as React from 'react';

import { TTCard } from './styles';

import { ITheme } from '../../types/theme';
import { ICard } from './types';

interface IProps {
  theme: ITheme;
  className?: string;
  card: ICard;
}

export const Card: React.SFC<IProps> = ({ card: { imageMobile, imageDesktop } }) => {
  return <TTCard className={'tt-card'} imageMobile={imageMobile} imageDesktop={imageDesktop} />;
};
