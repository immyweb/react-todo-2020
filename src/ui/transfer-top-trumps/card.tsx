import * as React from 'react';

import { Gradient } from './gradient';
import {
  CardContainer,
  CardHeader,
  CardInfo,
  CardLabel,
  CardProfileImage,
  Badge,
  BadgeImageHolder,
  CardFirstname,
  CardSurname,
  CardContent,
  CardClub,
  CardPlayer,
  CardStats,
  Divider,
  BadgeImage,
} from './styles';
import { colours } from '../../styles/colours';

import { ITheme } from '../../types/theme';
import { IPlayer, ICard } from './types';

interface IProps extends ICard {
  player: IPlayer;
  theme: ITheme;
  className?: string;
}

export const Card: React.SFC<IProps> = ({ currentClub, pos, estFee, playerAge, player, theme }) => {
  const {
    fname,
    sName,
    imageMobile,
    imageDesktop,
    club,
    clubBadge,
    age,
    fee,
    position,
    voted,
  } = player;

  return (
    <div>
      <CardContainer voted={voted}>
        <CardHeader className={'ttt-card-header'}>
          <CardFirstname fontColour={theme.primary}>{fname}</CardFirstname>
          <CardSurname fontColour={colours.$white}>{sName}</CardSurname>
        </CardHeader>
        <CardProfileImage
          className={'ttt-card-profileimage'}
          imageMobile={imageMobile}
          imageDesktop={imageDesktop}
        />
        <CardInfo className={'ttt-card-info'}>
          <BadgeImageHolder>
            <Gradient colour={colours.$white} height={4} top={17} opacity={0.5} />
            <Badge>{clubBadge && <BadgeImage src={clubBadge} alt={currentClub} />}</Badge>
          </BadgeImageHolder>
          <CardStats className={'ttt-card-stats'}>
            <CardClub>
              <CardLabel color={theme.primary}>{currentClub}</CardLabel>{' '}
              <CardContent>{club}</CardContent>
            </CardClub>
            <Divider height={41} mobileHide={true} />
            <CardPlayer>
              <CardLabel color={theme.primary}>{pos}</CardLabel>{' '}
              <CardContent>{position}</CardContent>
            </CardPlayer>
            <Divider height={41} />
            <CardPlayer>
              <CardLabel color={theme.primary}>{estFee}</CardLabel> <CardContent>{fee}</CardContent>
            </CardPlayer>
            <Divider height={41} />
            <CardPlayer>
              <CardLabel color={theme.primary}>{playerAge}</CardLabel>{' '}
              <CardContent>{age}</CardContent>
            </CardPlayer>
          </CardStats>
        </CardInfo>
      </CardContainer>
    </div>
  );
};
