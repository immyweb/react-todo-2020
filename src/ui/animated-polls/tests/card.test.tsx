import * as React from 'react';
import { shallow } from 'enzyme';

import { CardScreen } from '../card';

const card = {
  imageMobile: 'https://www.thesun.co.uk/wp-content/uploads/2018/10/handlebar.jpg',
  imageDesktop: '',
  title: 'The Chevron',
  copy: 'Looks a bit more Bruce Forsyth than Tom Selleck...',
};

function shallowRender(props?: any) {
  return shallow(<CardScreen {...card} {...props} />);
}

describe('<CardScreen />', () => {
  it('renders correctly', () => {
    const wrapper = shallowRender();
    expect(wrapper).toMatchSnapshot();
  });
});
