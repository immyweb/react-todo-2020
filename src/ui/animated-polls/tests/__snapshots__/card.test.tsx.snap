// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardScreen /> renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    copy="Looks a bit more Bruce Forsyth than Tom Selleck..."
    imageDesktop=""
    imageMobile="https://www.thesun.co.uk/wp-content/uploads/2018/10/handlebar.jpg"
    title="The Chevron"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <glamorous(glamorous(div))
          imageDesktop=""
          imageMobile="https://www.thesun.co.uk/wp-content/uploads/2018/10/handlebar.jpg"
        />,
        <glamorous(glamorous(div))>
          <glamorous(glamorous(h2))>
            The Chevron
          </glamorous(glamorous(h2))>
          <glamorous(glamorous(glamorous.P))>
            Looks a bit more Bruce Forsyth than Tom Selleck...
          </glamorous(glamorous(glamorous.P))>
        </glamorous(glamorous(div))>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "imageDesktop": "",
          "imageMobile": "https://www.thesun.co.uk/wp-content/uploads/2018/10/handlebar.jpg",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <glamorous(glamorous(h2))>
              The Chevron
            </glamorous(glamorous(h2))>,
            <glamorous(glamorous(glamorous.P))>
              Looks a bit more Bruce Forsyth than Tom Selleck...
            </glamorous(glamorous(glamorous.P))>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "The Chevron",
            },
            "ref": null,
            "rendered": "The Chevron",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Looks a bit more Bruce Forsyth than Tom Selleck...",
            },
            "ref": null,
            "rendered": "Looks a bit more Bruce Forsyth than Tom Selleck...",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <glamorous(glamorous(div))
            imageDesktop=""
            imageMobile="https://www.thesun.co.uk/wp-content/uploads/2018/10/handlebar.jpg"
          />,
          <glamorous(glamorous(div))>
            <glamorous(glamorous(h2))>
              The Chevron
            </glamorous(glamorous(h2))>
            <glamorous(glamorous(glamorous.P))>
              Looks a bit more Bruce Forsyth than Tom Selleck...
            </glamorous(glamorous(glamorous.P))>
          </glamorous(glamorous(div))>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "imageDesktop": "",
            "imageMobile": "https://www.thesun.co.uk/wp-content/uploads/2018/10/handlebar.jpg",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <glamorous(glamorous(h2))>
                The Chevron
              </glamorous(glamorous(h2))>,
              <glamorous(glamorous(glamorous.P))>
                Looks a bit more Bruce Forsyth than Tom Selleck...
              </glamorous(glamorous(glamorous.P))>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "The Chevron",
              },
              "ref": null,
              "rendered": "The Chevron",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Looks a bit more Bruce Forsyth than Tom Selleck...",
              },
              "ref": null,
              "rendered": "Looks a bit more Bruce Forsyth than Tom Selleck...",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
